src/plugins/discover/public/build_services.ts

```diff
@@ -75,6 +77,7 @@ export function buildServices(
   };
   const savedObjectService = createSavedSearchesLoader(services);
   const { usageCollection } = plugins;
+  const storage = new Storage(localStorage);
```

Comment:
I am not entirely understanding why we're trying to add this globally to the build_services. Since this is only used inside the DiscoverLayout component, wouldn't it make more sense to have the store just internally there? cc @kertal

Reply:
I would prefer having it here, in case we want to add more stuff to it.

Reply:
I think it makes sense to place it in services, because I can think of other settings that would benefit from local storage. For instance there was a recent request to remember the last state of hiding/showing the chart, unless it's persisted. This could be a nice follow up UX improvement. Any objections @timroes ?

Reply:
Nope, sounds good to me. I expected something like that is the reason, just wanted to double check :)